### PR TITLE
Make testsuite more consistent

### PR DESCRIPTION
**Pull request by malept**
It was hard to see the actual test failures due to the state of the testsuite. This makes sure all of the relational databases/tables are created correctly, and that some of the more obviously intermittent errors in the testcases are fixed.

### CODE REVIEW

File: test/elastic_record/integration/active_record_test.rb [hunk 1]
```diff
@@ -48,7 +48,7 @@ def setup
     setup_project_database(
       'adapter'   => 'mysql2',
       'host'      => "localhost",
-      'database'  => 'elastic_record_test',
```
Comment (matthuhiggins) [inline test/elastic_record/integration/active_record_test.rb]: why subtest?

File: test/elastic_record/relation_test.rb [hunk 2]
```diff
@@ -28,12 +28,14 @@ def test_to_hits
...
 
   def test_to_ids
+    Widget.elastic_index.delete_all
     create_widgets [Widget.new(id: 5, color: 'red'), Widget.new(id: 10, color: 'blue')]
 
     assert_equal ['5', '10'].to_set, Widget.elastic_relation.to_ids.to_set
   end
 
   def test_to_a
+    Widget.elastic_index.delete_all
```
Comment (matthuhiggins) [inline test/elastic_record/relation_test.rb]: :-1: This should not be necessary

**Comment by matthuhiggins** [comment]
mostly good

File: test/elastic_record/integration/active_record_test.rb (hunk 1 above)
Comment (malept) [inline test/elastic_record/integration/active_record_test.rb]: Seems like a bad idea to name it the same as the database you're using for the rest of the tests, if you're dropping/creating it within specific testcases.

File: test/elastic_record/relation_test.rb (hunk 2 above)
Comment (malept) [inline test/elastic_record/relation_test.rb]: I agree, but these tests assume that the indexes are empty at the beginning, and in running this testsuite over and over and over, this is only the case sometimes.

File: test/elastic_record/integration/active_record_test.rb (hunk 1 above)
Comment (matthuhiggins) [inline test/elastic_record/integration/active_record_test.rb]: Ah, this code runs after the test suite starts

File: test/elastic_record/relation_test.rb (hunk 2 above)
Comment (matthuhiggins) [inline test/elastic_record/relation_test.rb]: Been running these tests and have not seen this test fail.

Comment (matthuhiggins) [inline test/elastic_record/relation_test.rb]: Hah, 5 minutes later, I saw this test fail.

Comment (malept) [inline test/elastic_record/relation_test.rb]: I am happy to implement a different solution to fix these intermittent failures.

Comment (matthuhiggins) [inline test/elastic_record/relation_test.rb]: We do not have these failures in our apps, which do more complex things than here. I believe that is enough evidence to show that something is going wrong here.

This gem was originally made without a test/dummy app. I am thinking of changing the test suite to use real active_record models found in test/dummy/app/models.

Comment (ebarendt) [inline test/elastic_record/relation_test.rb]: We currently have two randomly failing tests. One in content_system is skipped. I think the other one is also in content_system.

Comment (matthuhiggins) [inline test/elastic_record/relation_test.rb]: That's due to scoring and ordering, which is different than these failures, which are due to bad persistence state when the test starts.

Comment (malept) [inline test/elastic_record/relation_test.rb]: I propose to merge this PR so we can see (real) test failures better, then you can delete my hacks once you change the testsuite to use the test/dummy app pattern.

File: test/elastic_record/index/mapping_test.rb [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 class ElasticRecord::Index::MappingTest < MiniTest::Test
   def test_delete_mapping
-    index_name = index.create
+    index_name = index.create("widgets_#{SecureRandom.random_number(100_000_000_000_000)}")
```
Comment (matthuhiggins) [inline test/elastic_record/index/mapping_test.rb]: Hardcode a name that is not used elsewhere, rather than random.

Comment (malept) [inline test/elastic_record/index/mapping_test.rb]: 5cba5487354f5e1b0513ddc7ce6753c4f7ff9bad

**Comment by matthuhiggins** [comment]
Looks good @malept. I will do a comprehensive test suite update during my post-xmas/pre-new-years week,

**Pull request merged**